### PR TITLE
16148: send reports from the submissions directly to FHIR Convert

### PR DESCRIPTION
**Pull request by mkalish**
This PR updates the pipeline so reports from the submission service are handled directly by the convert function

Test Steps:
1. Turn on the submission app
2. Post a report to it with valid sender
3. Verify the convert step picks it up

## Changes
- Update the convert step to handle a queue message from the submissions app or the receive function
- Updates the ReportStreamEventService to support queuing messages and then triggering them at a later point

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #16148 

## To Be Done
- Ticket to refactor ProcessFhirCommands

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/azure/FHIRFunctions.kt [hunk 1]
```diff
@@ -49,7 +51,15 @@ class FHIRFunctions(
         logger.info(
             "message consumed from elr-fhir-receive queue"
         )
-        process(message, dequeueCount, FHIRReceiver(), ActionHistory(TaskAction.receive))
+        process(message, dequeueCount, FHIRConverter(), ActionHistory(TaskAction.convert))
```
Comment (adegolier) [inline prime-router/src/main/kotlin/fhirengine/azure/FHIRFunctions.kt:54]: The function name is receive-fhir, but it's the convert action...?

Comment (mkalish) [reply prime-router/src/main/kotlin/fhirengine/azure/FHIRFunctions.kt:54]: I still need to clean this up.... but I'll be changing function name

File: prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt [hunk 2]
```diff
@@ -1023,4 +1038,116 @@ class FhirPathCommand : CliktCommand(
         stringValue.append("\n}\n")
         return stringValue.toString()
     }
+}
+
+// This exists only because ProcessFhirCommands instantiates a FHIRConverter to access a function that likely could be
+// made static
+// TODO: ticket to refactor it to make it static
```
[the remark this answers is not in the document]
Comment (mkalish) [reply prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt:1045]: https://github.com/CDCgov/prime-reportstream/issues/16407 and linked in the PR

File: prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt [hunk 3]
```diff
@@ -388,12 +401,15 @@ class ProcessFhirCommands : CliktCommand(
...
         }
-        val result = FHIRReceiverFilter().evaluateObservationConditionFilters(
-            receiver,
-            bundle,
-            ActionLogger(),
-            trackingId
-        )
+        val result =
+            FHIRReceiverFilter(
+                reportStreamEventService = NoopReportStreamEventService()
```
[the remark this answers is not in the document]
Comment (arnejduranovic) [reply prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt:406]: Reviewed ticket and made some updates, looks good to me!

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**